swift/ql/test/query-tests/Security/CWE-825/withUnsafeBytes.swift

```diff
@@ -0,0 +1,38 @@
+// stubs
+
+
+
+// tests
+
+func arrayTest() {
+    let ints = [1,2,3]
+    ints.withUnsafeBytes{ // BAD
+        return $0
+    }
+
+    print(ints.withUnsafeBytes({(p: UnsafeRawBufferPointer) -> UnsafeRawBufferPointer in return p})) // BAD
+
+    print(ints.withUnsafeBytes({(p: UnsafeRawBufferPointer) -> UnsafeRawBufferPointer in return id(pointer: p)})) // BAD
+
+    ints.withUnsafeBytes({(p: UnsafeRawBufferPointer) in print(p)}) // GOOD
+
+    var v = PointerHolder()
+    ints.withUnsafeBytes({(p: UnsafeRawBufferPointer) in v.field = p}) // BAD
+    print(v.field)
+
+    ints.withUnsafeBytes(myPrint) // GOOD
```

Comment:
Another variant here would be calling a member function that stores the pointer argument in the object it's called on for later use.  The effect is similar to the `v.field = p` case.